lib/db/db.rb

```diff
@@ -9,26 +9,63 @@ def size
     all.size
   end
 
+  def all
+    reader.nil? ? [] : reader.run
+  end
+
+  def at(index)
+    empty || invalid(index) ? {} : all[index]
```

Comment:
That's hard too read and not really idiomatic ruby (the ternary operator).
This method could actually read `all.fetch(index, {})`.